pkg/util/validation/validate.go

```diff
@@ -373,7 +373,7 @@ type MetadataValidationConfig interface {
 }
 
 // CleanAndValidateMetadata returns an err if a metric metadata is invalid.
-func CleanAndValidateMetadata(m *MetadataValidationMetrics, cfg MetadataValidationConfig, userID string, metadata *mimirpb.MetricMetadata) error {
+func CleanAndValidateMetadata(m *MetadataValidationMetrics, cfg MetadataValidationConfig, userID string, metadata *mimirpb.MetricMetadata) ValidationError {
```

Comment:
Why this change?

Reply:
@colega  `CleanAndValidateMetadata` returns only `ValidationError` errors, as expected, so I made it compliant with other validation methods from the package, that already return a `ValidationError`. This way, I could use the errors returned by all these methods inside the `NewValidationDistributorPushError()` constructor, that requires a `ValidationError` parameter.
If you don't agree with this change, I can change the signature of `NewValidationDsitributorPushError()`.

Reply:
If you revert this change the code still compiles and works in the same way. This is because this interface doesn't add any extra information to the error: the interfaces in Go define behaviors and are implicitly implemented.

Reply:
@colega I didn't get whether it is OK for you to do this change or you would prefer to revert it?

Reply:
I would revert this change (and not related to this PR, but would also remove the `ValidationError` type at all) as it doesn't do anything. 

You don't really need to change the signature of `NewValidationDistributorPushError` because both `error` and `validation.ValidationError` describe the same thing: something with `Error() string` method.

Reply:
OK, I have reverted it.